apps/wing/src/commands/test/snapshots.test.ts

```diff
@@ -0,0 +1,214 @@
+import { generateTmpDir } from "src/util";
+
+const prevCI = process.env.CI;
+
+describe("determineSnapshotMode", () => {
+  beforeEach(() => {
+    delete process.env.CI;
+  });
+
+  afterEach(() => {
+    process.env.CI = prevCI;
+  });
+
+  test("--snapshots=never", () => {
+    expect(
+      determineSnapshotMode("tf-aws", {
+        clean: false,
+        platform: [],
+        snapshots: SnapshotMode.NEVER,
+      })
+    ).toBe(SnapshotMode.NEVER);
+  });
+
+  test("--snapshots=dry", () => {
+    expect(
+      determineSnapshotMode("tf-aws", {
+        clean: false,
+        platform: [],
+        snapshots: SnapshotMode.UPDATE,
+      })
+    ).toBe(SnapshotMode.UPDATE);
+  });
+
+  test("--snapshots=deploy", () => {
+    expect(
+      determineSnapshotMode("tf-aws", {
+        clean: false,
+        platform: [],
+        snapshots: SnapshotMode.DEPLOY,
+      })
+    ).toBe(SnapshotMode.DEPLOY);
+  });
+
+  test("target=sim always disables (even if explicitly set)", () => {
+    expect(
+      determineSnapshotMode("sim", {
+        clean: false,
+        platform: [],
+        snapshots: SnapshotMode.DEPLOY,
+      })
+    ).toBe(SnapshotMode.NEVER);
+
+    expect(
+      determineSnapshotMode("sim", {
+        clean: false,
+        platform: [],
+        snapshots: SnapshotMode.ASSERT,
+      })
+    ).toBe(SnapshotMode.NEVER);
+
+    expect(
+      determineSnapshotMode("sim", {
+        clean: false,
+        platform: [],
+        snapshots: SnapshotMode.UPDATE,
+      })
+    ).toBe(SnapshotMode.NEVER);
+  });
+
+  describe("--snapshots=auto", () => {
+    describe("CI=1", () => {
+      beforeEach(() => {
+        process.env.CI = "1";
+      });
+
+      test("sim => disabled", () => {
+        expect(
+          determineSnapshotMode("sim", {
+            clean: false,
+            platform: [],
+          })
+        ).toBe(SnapshotMode.NEVER);
+      });
+
+      test("non sim => assert", () => {
+        expect(
+          determineSnapshotMode("tf-azure", {
+            clean: false,
+            snapshots: SnapshotMode.AUTO,
+            platform: [],
+          })
+        ).toBe(SnapshotMode.ASSERT);
+      });
+    });
+
+    describe("not in CI", () => {
+      test("sim => disabled", () => {
+        expect(
+          determineSnapshotMode("sim", {
+            clean: false,
+            platform: [],
+          })
+        ).toBe(SnapshotMode.NEVER);
+      });
+      test("non sim => deploy", () => {
+        expect(
+          determineSnapshotMode("tf-azure", {
+            clean: false,
+            platform: [],
+          })
+        ).toBe(SnapshotMode.DEPLOY);
+      });
+    });
+  });
+});
+
+describe("captureSnapshot", async () => {
+  async function makeEntrypoint() {
+    const workdir = await generateTmpDir();
+    const entrypoint = join(workdir, "main.w");
+
+    await writeFile(
+      entrypoint,
+      `
+      bring cloud;
+      new cloud.Bucket();
+    `
```

Comment:
It would be useful to see a `test {}` block in this example to get a more realistic file that someone may run `wing test` on.